packages/@aws-cdk/aws-cloudfront/lib/origin.ts

```diff
@@ -0,0 +1,194 @@
+import { IBucket } from '@aws-cdk/aws-s3';
+import { Construct } from '@aws-cdk/core';
+import { CfnDistribution } from './cloudfront.generated';
+import { OriginProtocolPolicy } from './distribution';
+import { OriginAccessIdentity } from './origin_access_identity';
+
+/**
+ * Properties to be used to create an Origin. Prefer to use one of the Origin.from* factory methods rather than
+ * instantiating an Origin directly from these properties.
+ *
+ * @experimental
+ */
+export interface OriginProps {
+  /**
+   * The domain name of the Amazon S3 bucket or HTTP server origin.
+   */
+  readonly domainName: string;
+}
+
+/**
+ * Options passed to Origin.bind().
+ */
+export interface OriginBindOptions {
```

Comment:
can this be kept outside the public API? (as an internal type)

Reply:
I didn't realize we could do that, given that bind() is public (albeit now marked internal). But it builds and compiles that way, so I guess we can?

Reply:
I think that if `_bind` is internal, then the types in it's signature do not have to be public because jsii won't try to resolve them.